packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -275,6 +284,27 @@ export class AmplifyGraphqlApi extends Construct {
     outputStorageStrategy.addBackendOutputEntry(graphqlOutputKey, output);
   }
 
+  /**
+   * Gets the deployment information from the context.
+   * @returns the deployment identifier.
+   */
+  private getDeploymentIdentifier(): DeploymentIdentifier {
+    const deploymentType = this.node.tryGetContext(CDKContextKey.DEPLOYMENT_TYPE);
+
+    if (deploymentType === 'sandbox') {
+      return {
+        deploymentType: deploymentType,
+        namespace: this.node.tryGetContext(CDKContextKey.BACKEND_NAMESPACE),
```

Comment:
What is the behavior if `BACKEND_NAMESPACE` context key isn't set? Will there be a meaningful error propagated to a customer-facing context?

..
..
..

Trick question :). Even if the error is something like "backendNamespace context key not set", that doesn't tell the customer how to fix it. We should catch the condition ourselves -- if either of the context keys are set, both must be set, and we should craft an appropriate error message to help customers resolve the situation.)